libsolidity/analysis/TypeChecker.cpp

```diff
@@ -627,6 +628,22 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 		}
 		return true;
 	});
+
+	int deposit = assembly.deposit();
```

Comment:
I think this should be a feature of the inline assembly module itself. The place would be `void CodeTransform::operator()(assembly::Block const& _block)` - there is even a comment about stack height.